stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerViewModelBinding.kt

```diff
@@ -0,0 +1,24 @@
+package io.getstream.chat.android.ui.message.composer
+
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.coroutineScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+public fun MessageComposerViewModel.bindView(view: MessageComposerView, lifecycleOwner: LifecycleOwner) {
+
+    view.onSendMessageAction = {
+        val text = input.value
+        val attachments = selectedAttachments.value
+        val message = buildNewMessage(text, attachments)
+        sendMessage(message)
+    }
+
+    view.onInputTextChanged = { setMessageInput(it) }
+
+    lifecycleOwner.lifecycle.coroutineScope.launch {
+        input.collect { input ->
+            view.messageInputState.value = view.messageInputState.value.copy(inputValue = input)
```

Comment:
This wouldn't be required if the state came from the ViewModel - could just be a single line to set the input for the ViewModel

Reply:
Yup, view model could expose `messageInputState`. Would it work with compose as well or should it be XML-specific?

Reply:
For now, I've exposed it in ui-components `MessageComposerViewModel` only.